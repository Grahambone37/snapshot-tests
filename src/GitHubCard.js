import React, { useState, useEffect } from 'react'
import { Card } from 'react-bootstrap'

export default function GitHubCard() {
    const [gitUsername, setGitUsername] = useState('username')
    const [gitImage, setGitImage] = useState('Git Image')
    const [gitBio, setGitBio] = useState('Git Bio')

    useEffect(() => {
        fetch('https://api.github.com/users/Grahambone37')
            .then(res => res.json())
            .then(data => {
                setGitUsername(data.name)
                setGitImage(data.avatar_url)
                setGitBio(data.bio)
            })
    }, [])

    return (
        <Card>
            <Card.Img src={gitImage} style={{
                height: '50px',
                width: '50px',
                marginLeft: '48%'
            }} />
            <h1>{gitUsername}</h1>
            <h2>{gitBio}</h2>
        </Card>
    )
}


