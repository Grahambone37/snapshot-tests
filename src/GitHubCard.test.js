// import App from './App';
// import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer'
import GitHubCard from './GitHubCard';

// tried to set this to wait for a while for the page to load, but gonna move on
test('Github profile snapshot', async () => {
    const tree = renderer.create(<GitHubCard />).toJSON()
    expect(tree).toMatchSnapshot()
})