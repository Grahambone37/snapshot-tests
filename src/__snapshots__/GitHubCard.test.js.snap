// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github profile snapshot 1`] = `
<div
  className="card"
>
  <img
    className="card-img"
    src="Git Image"
    style={
      Object {
        "height": "50px",
        "marginLeft": "48%",
        "width": "50px",
      }
    }
  />
  <h1>
    username
  </h1>
  <h2>
    Git Bio
  </h2>
</div>
`;
